### PR TITLE
CORE-8432 filtered tx impl

### PR DESCRIPTION
**Pull request by blsemo**
Implementation for the UtxoFilteredTransaction interface

Make sure to merge https://github.com/corda/corda-api/pull/719 first and reference the correct API in this PR before  merging.

### CODE REVIEW

File: components/ledger/ledger-utxo-flow/src/integrationTest/kotlin/net/corda/ledger/utxo/flow/impl/transaction/serializer/tests/UtxoFilteredTransactionAMQPSerializationTest.kt [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        assertThat(inputs.size).isEqualTo(2)
+        assertThat(inputs.values.size).isEqualTo(2)
+        assertThat(inputs.values[0]?.transactionHash).isEqualTo(inputHash)
+
+        assertThat(deserialized.outputStateAndRefs).isInstanceOf(UtxoFilteredData.SizeOnly::class.java)
+        val outputs = deserialized.outputStateAndRefs as UtxoFilteredData.SizeOnly
+        assertThat(outputs.size).isEqualTo(2)
+    }
+
+    data class MyState(val value: Int) : ContractState {
```
Comment (vlajos) [inline components/ledger/ledger-utxo-flow/src/integrationTest/kotlin/net/corda/ledger/utxo/flow/impl/transaction/serializer/tests/UtxoFilteredTransactionAMQPSerializationTest.kt:88]: I think we have a UtxoStateClassExample.
But it is not too important.

File: libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/amqp/LocalSerializerFactory.kt [hunk 2]
```diff
@@ -272,8 +272,18 @@ class DefaultLocalSerializerFactory(
...
+                resolved.keyType.observedType.asClass(),
+                resolved.valueType.observedType.asClass()
+            )
+            override fun getRawType(): Type = resolved.observedType.asClass()
+            override fun getOwnerType(): Type? = (resolved.typeIdentifier as? ParameterizedType)?.ownerType
+        }
+
         return makeAndCache(resolved) {
-            MapSerializer(resolved.typeIdentifier.getLocalType(sandboxGroup) as ParameterizedType, this)
+            MapSerializer(locallyDeclaredType, this)
```
Comment (vlajos) [inline libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/amqp/LocalSerializerFactory.kt:286]: What does this change do?

**Review by vlajos — DISMISSED** [review]
LGTM

File: gradle.properties [hunk 3]
```diff
@@ -40,7 +40,8 @@ bouncycastleVersion=1.70
 # Corda API libs revision (change in 4th digit indicates a breaking change)
 # Change to 5.0.0.xx-SNAPSHOT to pick up maven local published copy
 #cordaApiVersion=5.0.0.xxx-SNAPSHOT
-cordaApiVersion=5.0.0.511-beta+
+#cordaApiVersion=5.0.0.510-beta+
+cordaApiVersion=5.0.0.510-alpha-1670317162561
```
Comment (driessamyn) [inline gradle.properties]: Be careful not to merge it like this.

File: libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/amqp/LocalSerializerFactory.kt (hunk 2 above)
Comment (lankydan) [reply libs/serialization/serialization-amqp/src/main/kotlin/net/corda/internal/serialization/amqp/LocalSerializerFactory.kt:286]: Fixed a serialization error where a type couldn't be found in the sandbox.

**Review by lankydan — APPROVED** [review]

**Pull request merged**